src/test/PolygonStateBridge.t.sol

```diff
@@ -0,0 +1,196 @@
+
+    uint32 public opGasLimit;
+
+    address public mockWorldIDAddress;
+
+    address public fxRoot;
+    address public checkpointManager;
+    address public owner;
+    uint256 public sampleRoot;
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            EVENTS                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice OpenZeppelin Ownable.sol transferOwnership event
+    /// @param previousOwner The previous owner of the StateBridge contract
+    /// @param newOwner The new owner of the StateBridge contract
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    /// @notice OpenZeppelin Ownable2Step transferOwnership event
+    /// @param previousOwner The previous owner of the StateBridge contract
+    /// @param newOwner The new owner of the StateBridge contract
+    event OwnershipTransferStarted(address indexed previousOwner, address indexed newOwner);
+
+    /// @notice Emmitted when the the StateBridge sets the root history expiry for OpWorldID and PolygonWorldID
+    /// @param rootHistoryExpiry The new root history expiry
+    event SetRootHistoryExpiry(uint256 rootHistoryExpiry);
+
+    /// @notice Emmitted when a root is sent to PolygonWorldID
+    /// @param root The latest WorldID Identity Manager root.
+    event RootPropagated(uint256 root);
+
+    function setUp() public {
+        /// @notice Create a fork of the Ethereum mainnet
+        mainnetFork = vm.createFork(MAINNET_RPC_URL);
+
+        vm.selectFork(mainnetFork);
+        /// @notice Roll the fork to a block where FXPortal already has been deployed
+        /// @notice and the Base crossDomainMessenger ResolvedDelegateProxy target address is initialized
+        vm.rollFork(17711915);
+
+        sampleRoot = uint256(0x123);
+        mockWorldID = new WorldIDIdentityManagerMock(sampleRoot);
+        mockWorldIDAddress = address(mockWorldID);
+
+        checkpointManager = address(0x86E4Dc95c7FBdBf52e33D563BbDB00823894C287);
+        fxRoot = address(0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2);
+
+        polygonStateBridge = new PolygonStateBridge (
+            checkpointManager,
+            fxRoot,
+            mockWorldIDAddress
+        );
+
+        owner = polygonStateBridge.owner();
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                           SUCCEEDS                          ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice select a specific fork
+    function test_canSelectFork_succeeds() public {
+        // select the fork
+        vm.selectFork(mainnetFork);
+        assertEq(vm.activeFork(), mainnetFork);
+    }
+
+    /// @notice tests that a root can be sent successfully to Polygon
+    function test_propagateRoot_succeeds(address randomAddress) public {
```

Comment:
We should add a test for the failure cases for `propagateRoot`

Reply:
done